core-java/src/main/java/eu/ostrzyciel/jelly/core/ProtoEncoder.java

```diff
@@ -0,0 +1,57 @@
+package eu.ostrzyciel.jelly.core;
+
+import eu.ostrzyciel.jelly.core.internal.ProtoEncoderBase;
+import eu.ostrzyciel.jelly.core.internal.RowBufferAppender;
+import eu.ostrzyciel.jelly.core.proto.v1.Rdf;
+import java.util.List;
+
+public abstract class ProtoEncoder<TNode, TTriple, TQuad>
+    extends ProtoEncoderBase<TNode, TTriple, TQuad>
+    implements RowBufferAppender {
+
+    public record Params(
+        Rdf.RdfStreamOptions options,
+        boolean enableNamespaceDeclarations,
+        List<Rdf.RdfStreamRow> appendableRowBuffer
```

Comment:
We only ever use the `add(E e)` method. So maybe make this arg into `Collection<RdfStreamRow>`?

Reply:
doned